src/shell/Program.cs

```diff
@@ -1,92 +1,154 @@
+﻿using System.Diagnostics;
+using System.IO.Abstractions;
 using System.Runtime.InteropServices;
 using McMaster.Extensions.CommandLineUtils;
 using Microsoft.Extensions.DependencyInjection;
 using NeoShell.Commands;
-
+using NeoShell.Models;
+using Newtonsoft.Json;
 
 namespace NeoShell
 {
-  [Command("neosh", Description = "Neo N3 blockchain shell", UsePagerForHelpText = false)]
-  [Subcommand(typeof(ContractCommand))]
-  [Subcommand(typeof(ConnectCommand))]
-  [Subcommand(typeof(ShowCommand))]
-  [Subcommand(typeof(TransferCommand))]
-  class Program
-  {
-    public static async Task<int> Main(string[] args)
+    [Command("neosh", Description = "Neo N3 blockchain shell", UsePagerForHelpText = false)]
+    [Subcommand(typeof(ContractCommand))]
+    [Subcommand(typeof(ConnectCommand))]
+    [Subcommand(typeof(ShowCommand))]
+    [Subcommand(typeof(TransferCommand))]
+    class Program
     {
-      EnableAnsiEscapeSequences();
-
-      var services = new ServiceCollection()
-          .AddSingleton<ExpressChainManagerFactory>()
-          .AddSingleton<TransactionExecutorFactory>()
-          .AddSingleton<IFileSystem, FileSystem>()
-          .AddSingleton<IConsole>(PhysicalConsole.Singleton)
-          .BuildServiceProvider();
-
-      var app = new CommandLineApplication<Program>();
-      app.Option<bool>("--stack-trace", "", CommandOptionType.NoValue, o => o.ShowInHelpText = false, inherited: true);
-      app.Conventions
-          .UseDefaultConventions()
-          .UseConstructorInjection(services);
-
-      try
-      {
-        return await app.ExecuteAsync(args);
-      }
-      catch (CommandParsingException ex)
-      {
-        await Console.Error.WriteLineAsync($"\x1b[1m\x1b[31m\x1b[40m{ex.Message}");
-
-        if (ex is UnrecognizedCommandParsingException uex && uex.NearestMatches.Any())
+        public static async Task<int> Main(string[] args)
         {
-          await Console.Error.WriteLineAsync();
-          await Console.Error.WriteLineAsync("Did you mean this?");
-          await Console.Error.WriteLineAsync("    " + uex.NearestMatches.First());
+            EnableAnsiEscapeSequences();
+
+            var services = new ServiceCollection()
+                .AddSingleton<ExpressChainManagerFactory>()
+                .AddSingleton<TransactionExecutorFactory>()
+                .AddSingleton<IFileSystem, FileSystem>()
+                .AddSingleton<IConsole>(PhysicalConsole.Singleton)
+                .BuildServiceProvider();
+
+            var app = new CommandLineApplication<Program>();
+            app.Option<bool>("--stack-trace", "", CommandOptionType.NoValue, o => o.ShowInHelpText = false, inherited: true);
+            app.Conventions
+                .UseDefaultConventions()
+                .UseConstructorInjection(services);
+            ShellExtensions extensions = LoadShellExtensions(services.GetService<IFileSystem>());
+            try
+            {
+                ShellExtension? extension;
```

Comment:
inline declaration

Reply:
This is cool. I guess my C# knowledge is stale (stuck at 6.0)

Reply:
I should have noted this was a nit pick